### PR TITLE
Add guild membership

### LINKED ISSUE #40: Track guild membership
Right now we don't really have a good way of checking if a person is a member of a guild.
You _can_ lookup in `role_membership` when the guild ID matches joining on the `roles` table to find the `@everyone` role, but that's a huge pain, and since role information is preserved even for people who leave, it's not totally accurate.

So instead a separate table for tracking guild-specific data should be made.

### PR DESCRIPTION
**Pull request by ammongit**
Fixes #40 

This replaces the `nicknames` table with a single table that tracks both nicks and guild membership, when they joined, and their current membership status as a boolean.

### CODE REVIEW

File: statbot/sql.py [hunk 1]
```diff
@@ -972,6 +972,7 @@ def remove_member(self, trans, member):
         # Don't delete role membership
 
     def remove_old_members(self, trans, guild):
+        # pylint: disable=singleton-comparison
```
Comment (robstolarz) [inline statbot/sql.py]: Can you leave a note as to why this is here? I don't like seeing random pylint suppressions, haha

Comment (ammongit) [reply statbot/sql.py]: It complains about the `self.tb_guild_membership.c.is_member == True`.

Comment (robstolarz) [reply statbot/sql.py]: Hm, I see. Are we really concerned that `self.tb_guild_membership.c.is_member` could return a non-bool?

Comment (ammongit) [reply statbot/sql.py]: Well it's not a python `bool`, it's a weird SQLAlchemy column/type/schema something object. When you do `table.c.column == value`, it generates an "equality" object, which it uses to create the `WHERE` clause of the query.

Comment (robstolarz) [reply statbot/sql.py]: Can you amend your comment to include that? That is super instructive.

**Pull request merged**